packages/page/src/UserInterface/Command/InitializeHomepageCommand.php

```diff
@@ -0,0 +1,126 @@
+use Sulu\Component\Webspace\Webspace;
+use Sulu\Content\Domain\Model\WorkflowInterface;
+use Sulu\Messenger\Infrastructure\Symfony\Messenger\FlushMiddleware\EnableFlushStamp;
+use Sulu\Page\Application\Message\ApplyWorkflowTransitionPageMessage;
+use Sulu\Page\Application\Message\CreatePageMessage;
+use Sulu\Page\Application\MessageHandler\CreatePageMessageHandler;
+use Sulu\Page\Domain\Model\PageInterface;
+use Sulu\Page\Domain\Repository\PageRepositoryInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\HandleTrait;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+/**
+ * @internal Your code should not create direct dependencies on this implementation
+ *  projects can utilize the `sulu:page:initialize` command to
+ *  initialize new webspaces with the homepage
+ *
+ * @final
+ */
+#[AsCommand(name: 'sulu:page:initialize', description: 'Initializes the homepage per webspace')]
+class InitializeHomepageCommand extends Command
+{
+    use HandleTrait;
+
+    public function __construct(
+        private WebspaceManagerInterface $webspaceManager,
+        private PageRepositoryInterface $pageRepository,
+        MessageBusInterface $messageBus
+    ) {
+        parent::__construct();
+        $this->messageBus = $messageBus;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $ui = new SymfonyStyle($input, $output);
+
+        $webspaces = $this->webspaceManager->getWebspaceCollection();
+        foreach ($webspaces as $webspace) {
+            $ui->section('Initialize homepage for webspace ' . $webspace->getKey());
+
+            $localizations = $webspace->getLocalizations();
+            foreach ($localizations as $localization) {
+                if ($this->homepageExists($webspace, $localization->getLocale())) {
+                    $ui->text(
+                        \sprintf(
+                            'Homepage for locale "%s" in webspace "%s" already exists',
+                            $localization->getLocale(),
+                            $webspace->getKey()
+                        )
+                    );
+                    continue;
+                }
+                $homepage = $this->createHomepage($webspace, $localization->getLocale());
+                $this->publishHomepage($homepage->getUuid(), $localization->getLocale());
+            }
+        }
+        $ui->success('Homepage initialization completed');
+
+        return Command::SUCCESS;
+    }
+
+    private function homepageExists(Webspace $webspace, string $locale): bool
+    {
+        // TODO more optimized query
+        $homepage = $this->pageRepository->findOneBy([
```

Comment:
@alexander-schranz should we introduce a `countBy` or `exists` method to the interface? 🤔